baybe/constraints/discrete.py

```diff
@@ -59,6 +72,22 @@ def get_invalid(self, data: pd.DataFrame) -> pd.Index:  # noqa: D102
 
         return data.index[mask_bad]
 
+    def get_polars_expression(self, df: pl.LazyFrame) -> pl.Expr:
+        """Translate the conditions to polars expression for filtering.
+
+        Args:
+            df: A polars lazy frame of all parameters.
+
+        Returns:
+            The Polars Expr object to pass as an argument to filter().
+
+        """
+        op = self.condition.convert_to_polars()
+        cols = [c for c in self.parameters if c in df.columns]
+        if set(cols) == set(self.parameters):
+            return op(pl.sum_horizontal(cols))
+        return pl.lit(True)
```

Comment:
why the pl.lit return? seems like that would silently brush over the fact that something unintended happened

Reply:
I substituted  `return None` with this line to fix mypy errors

Reply:
filter(pl.lit(True)) has no effect on a Polars dataframe. 
I'm doing this because we do **partial** products, and many times all the constraint's columns don't exist in this partial result. So in such situations we want to make sure nothing is omitted.

Reply:
i didnt realize youre doing the filtering repeatdely on partial parts of the product
Is there a reason for it? Seems like a very strange design choice otherwise

Reply:
I think one of the reasons we switched to Polars was that it allowed us to filter out undesired rows **during** the product operation, hence reducing dataframe size as we go. For example, as soon as all the paramters of a sum constraint become available, we check if it violates the constraint or not.
Please correct me if I'm wrong.

Reply:
I explained the computational difference [here](https://github.com/emdgroup/baybe/pull/252/files#r1633637429)